src/Paprika/Merkle/RlpMemo.cs

```diff
@@ -64,139 +57,147 @@ public bool TryGetKeccak(byte nibble, out ReadOnlySpan<byte> keccak)
         return false;
     }
 
-    private Span<byte> GetAtNibble(byte nibble) => _buffer.Slice(nibble * Keccak.Size, Keccak.Size);
-
-    public static RlpMemo Decompress(scoped in ReadOnlySpan<byte> leftover, NibbleSet.Readonly children,
-        scoped in Span<byte> workingSet)
+    public bool Exists(byte nibble)
     {
-        if (_decompressionForbidden)
+        if (_buffer.Length == 0)
         {
-            ThrowDecompressionForbidden();
+            return false;
         }
 
-        var span = workingSet[..Size];
+        GetIndex(out var index);
+
+        return index[nibble];
+    }
 
-        if (leftover.IsEmpty)
+    private Span<byte> GetAtNibble(byte nibble)
+    {
+        if (_buffer.Length == 0)
         {
-            // no RLP cached yet
-            span.Clear();
-            return new RlpMemo(span);
+            return [];
         }
 
-        if (leftover.Length == Size)
+        GetIndex(out var index);
+
+        // Check if the element exists
+        if (!index[nibble])
         {
-            leftover.CopyTo(span);
-            return new RlpMemo(span);
+            return [];
         }
 
-        // It's neither empty nor full. It must be the compressed form, prepare setup first
-        span.Clear();
-        var memo = new RlpMemo(span);
+        var nibbleIndex = index.SetCountBefore(nibble) - 1;
+        return _buffer.Slice(nibbleIndex * Keccak.Size, Keccak.Size);
+    }
 
-        // Extract empty bits if any
-        NibbleSet.Readonly empty;
+    private void GetIndex(out NibbleSet.Readonly index)
```

Comment:
The `NibbleSet.Readonly` is only 2 bytes long. Have you considered returning it by value?